tools/include/eosio/abigen.hpp

```diff
@@ -96,12 +92,8 @@ namespace eosio { namespace cdt {
          auto action_name = decl->getEosioActionAttr()->getName();
 
          if (!checked_actions.insert(get_action_name(decl)).second)
-            CDT_CHECK_WARN(!rcs[get_action_name(decl)].empty(), "abigen_warning", decl->getLocation(), "Action <"+get_action_name(decl)+"> does not have a ricardian contract");
-
-         if (!suppress_ricardian_warnings)
-            if (rcs[get_action_name(decl)].empty())
-               // TODO
-               std::cout << "Warning, action <"+get_action_name(decl)+"> does not have a ricardian contract\n";
+            if (!suppress_ricardian_warnings)
+               CDT_CHECK_WARN(!rcs[get_action_name(decl)].empty(), "abigen_warning", decl->getLocation(), "Action <"+get_action_name(decl)+"> does not have a ricardian contract");
```

Comment:
what is the difference? is it now outputs to stderr instead of stdout?

Reply:
Duplicated code. It seems original author intended replacing stdout warning message with CDT_CHECK_WARN macro when macro is implemented, but just added rather than replace it.